packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -134,9 +136,10 @@ class FlutterDriverExtension {
       'tap': (Map<String, String> params) => Tap.deserialize(params),
       'waitFor': (Map<String, String> params) => WaitFor.deserialize(params),
       'waitForAbsent': (Map<String, String> params) => WaitForAbsent.deserialize(params),
-      'waitUntilNoTransientCallbacks': (Map<String, String> params) => WaitUntilNoTransientCallbacks.deserialize(params),
-      'waitUntilNoPendingFrame': (Map<String, String> params) => WaitUntilNoPendingFrame.deserialize(params),
-      'waitUntilFirstFrameRasterized': (Map<String, String> params) => WaitUntilFirstFrameRasterized.deserialize(params),
+      'waitForCondition': (Map<String, String> params) => WaitForCondition.deserialize(params),
+      'waitUntilNoTransientCallbacks': (Map<String, String> params) => WaitUntilNoTransientCallbacks.deserialize(params), // ignore: deprecated_member_use_from_same_package
```

Comment:
Ditto on deprecation question

Reply:
If we replace the wait APIs like `waitUntilNoTransientCallbacks` with `waitForCondition`, this is going to be a hard-breaking change, as the JSON-RPC command for `waitUntilNoTransientCallbacks` is something like `{command: waitUntilNoTransientCallbacks}` and only the `WaitUntilNoTransientCallbacks` command class can understand/deserialize it.

Flutter Driver is updated to send the new `waitForCondition` command to this extension API, but we'll need to update Espresso/EarlGrey to use the new API before we remove it.